swan-lake/learn-the-platform/ballerina-persist/overview.md

```diff
@@ -0,0 +1,68 @@
+---
+layout: ballerina-persist-documentation-left-nav-pages-swanlake
+title: Bal Persist Overview
+description: The sections gives an overview of the bal persist feature.
+keywords: ballerina, programming language, ballerina packages, persist, persist model, persist cli, persist client api
+permalink: /learn/ballerina-persist/overview/
+active: persist_overview
+intro: The bal persist feature allows you to store data in different data stores and retrieve them when needed. A data store can be a database, a in-memory cache, or a file system. Ballerina supports the following data stores: In-memory table, MySQL, Google Sheets. The important point is you can use the same syntax to access data in all these data stores. So you don't need to learn different syntaxes to access data in different data stores.
+redirect_from:
+- /learn/ballerina-persist/overview
+---
+
+This feature has three main sections: Persist Model, Persist CLI and Persist Client API. The Persist Model is used to define the data model. The Persist CLI is used to generate the client API for the data model. The Persist Client API is used to access the data in the data store.
+
+## Persist Model
+
+The Persist Model is used to define the data model. The data model is defined in a Ballerina source file inside the persist directory in the project root directory. The data model is defined using the Ballerina record type. The following example shows how to define a data model for a table named `Employee` with the fields `id`, `name`, `age` and `salary`.
+
+```ballerina
+type Employee record {
+    readonly int id;
+    string name;
+    int age;
+    float salary;
+}
+```
+
+The `readonly` keyword is used to define a field as a primary key. You need to have at least one primary key field in the Entity record. The primary key field is used to uniquely identify a record in the table. The primary key field is used to generate the `get`, `update` and `delete` operations in the client API. You can define multiple primary key fields in the Entity record.
+
+Learn about how to define relationships between entity records and more about the persist model in the [Persist Model](persist-model.md) section.
+
+> Note: The Ballerina VSCode plugin provides validation and code actions for the persist model. So you can easily create the persist model using the Ballerina VSCode plugin.
+
+## Persist CLI
+
+The Persist CLI is used to generate the client API for the data model. Based on the data store, additional configurable file and setup scripts are generated. For example, if you are using MySQL as the data store, the `persist_db_config.bal` file and the `script.sql` script are generated. The `persist_db_config.bal` file is used to configure the MySQL database connection. The `script.sql` file is used to create the table in the MySQL database. The client API is generated in the `generated` directory in the project root directory.
```

Comment:
Add comma before and. 
`persist_db_config.bal` file, and the `script.sql` script are generated.`